pkg/kf/commands/routes/create_route.go

```diff
@@ -80,7 +80,7 @@ Instead use the --namespace flag.`)
 				},
 				ObjectMeta: metav1.ObjectMeta{
 					Namespace: space,
-					Name: v1alpha1.GenerateRouteName(
+					Name: v1alpha1.GenerateRouteClaimName(
```

Comment:
> we could create a webhook to block deletion of a claim if any routes belonged to it

Is this equivalent to making the RouteClaim an OwnerReference of a Route?

Reply:
I don't think it is quite. From my understanding K8s would allow someone to do the delete, and the delete would attempt to remove all the routes which the reconcilers would fight; unless the propagation policy was changed to just orphan the routes. I _think_ from there they'd be picked up during a later garbage collection cycle.

Reply:
I've added the RouteClaim type, I'll leave some of the other features for later PRs